### PR TITLE
service accounts

### PR DESCRIPTION
**Pull request by guirava**
We introduce here Service Accounts with the class `ServiceAccount` in `service_account.py`.
A `ServiceAccount` object represents a Polaris service account and can be created parametrically or from the JSON file the Polaris Service Account UI generates.
With a service account you can:
- obtain an access token with `ServiceAccount.get_client_token()`,
- or create a new session directly with `PolarisClient.from_service_account()`.
- Also, obtain an access token to reach the *Appliance* API at the cluster with `ServiceAccount.get_cluster_client_token()`

For example of use, see `sample/rubrik_polaris/get_cdm_cluster_client_token.py`

### CODE REVIEW

**Comment by pmilano1** [comment]
Can youu summarize the changes/features added? Thanks!

**Review by pmilano1 — COMMENTED** [review]
This looks ok to me - thanks for addressing changes. Will have @leitzler take a quick look also before merge.

File: rubrik_polaris/rubrik_polaris.py [hunk 1]
```diff
@@ -20,107 +20,141 @@
...
 
         # Switch off SSL checks if needed
-        if 'insecure' in self._kwargs and self._kwargs['insecure']:
+        if self._kwargs.get('insecure', False):
             urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
 
-        # Adjust Polaris domain if a custom root is defined
-        if 'root_domain' in self._kwargs and self._kwargs['root_domain'] is not None:
-            self._baseurl = "https://{}.{}/api".format(self._domain, self._kwargs['root_domain'])
+        if self._baseurl:
```
Comment (leitzler) [inline rubrik_polaris/rubrik_polaris.py]: Is there a reason to use the debug statement `assert` here over doing this?
```python
if self._baseurl and not self._access_token:
    raise [...]
```

File: sample/polaris_client.py [hunk 2]
```diff
@@ -2,34 +2,45 @@
...
+else:
+    ### Instantiate with username/password
+    rubrik = PolarisClient.from_credentials(
+        domain=args.domain, username=args.username,
+        password=args.password, root_domain=args.root_domain,
+        insecure=args.insecure)
+
+# except Exception as err:
+#     print(err)
+#     sys.exit(1)
```
Comment (leitzler) [inline sample/polaris_client.py]: Is this block left commented on purpose?

File: sample/rubrik_polaris/get_appliance_token.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
+"""Create a session token for a given appliance with a service account,
+and retrieve cluster version string from the appliance API.
+
+This sample script needs 3 pieces of information to run:
+ 1. the location of a service account JSON file (that you downloaded from
+    the UI when creating a service account);
+ 2. the unique identifier of the appliance (typically, the cluster UUID);
+ 3. the appliance node IP (typically, a cluster node IP address).
```
Comment (leitzler) [inline sample/rubrik_polaris/get_appliance_token.py]: Looks like this is fetched from Polaris so the entire `3. ... ` can be removed

**Review by leitzler — COMMENTED** [review]
Thank you for this PR and apologies for the delayed response, here are some comments & questions.

**Comment by pmilano1** [comment]
@guirava ,  we will be merging https://github.com/rubrikinc/rubrik-polaris-sdk-for-python/pull/31 prior to this, so you will need to rebase after that. Apologies for the delay in this.

**Comment by leitzler** [comment]
Hi @guirava, #31 is now merged. Can you please rebase & resolve the conflicts when you take a look at the comments I added earlier. Thanks!

File: rubrik_polaris/service_account.py [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+        :param appliance_uuid: Appliance or cluster UUID.
+               Example value:
+               '40505837-9772-4a91-a18a-db6108c66b59'
+        :return: tuple of 3 strings: session id, token and expiration date.
+
+        Response example: (client_id, client_token, expiration) with:
+            client_id = '009e7254-3e1b-4bef-a06a-6a0dc63cb6fe'
+            client_token = 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJmNjFk'\
+                    'N2RkNS1mN2M4LTQ3MDEtYWFiOS00ZGJmZDkxN2IwMmNfY2xpZW50fHJqU09M'\
```
Comment (PrashanthiMNaik) [inline rubrik_polaris/service_account.py]: :exclamation:**Cycode**: Secret of type: 'Base64 Entropy' was found.
Severity: Medium

File: rubrik_polaris/service_account.py [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+                    'ZGVuazdndEZXU25TaVNnWDRHMVNwcmRrRjZJIiwiaXNNZmFSZW1lbWJlclRv'\
```
Comment (PrashanthiMNaik) [inline rubrik_polaris/service_account.py]: :exclamation:**Cycode**: Secret of type: 'Base64 Entropy' was found.
Severity: Medium

File: rubrik_polaris/service_account.py [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+        - see list_cdm_clusters to retrieve your cluster ID.
+
+        :param appliance_uuid: Appliance or cluster UUID.
+               Example value:
+               '40505837-9772-4a91-a18a-db6108c66b59'
+        :return: tuple of 3 strings: session id, token and expiration date.
+
+        Response example: (client_id, client_token, expiration) with:
+            client_id = '009e7254-3e1b-4bef-a06a-6a0dc63cb6fe'
+            client_token = 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJmNjFk'\
```
Comment (PrashanthiMNaik) [inline rubrik_polaris/service_account.py]: :exclamation:**Cycode**: Secret of type: 'Generic Password' was found.
Severity: High

**Pull request closed without merging**